configure.ac

```diff
@@ -536,7 +538,9 @@ dnl Multi threaded library.
    AS_IF([test "x$ac_cv_atomic_dec_uint_nv" = "xyes"],
      [AC_DEFINE([LOG4CPLUS_HAVE_ATOMIC_DEC_UINT_NV])])
 
-   AX_TLS_SUPPORT
+   if test "$os_iphone" != "yes"; then
```

Comment:
I do not think that we need `ax_check_host_and_build_os.m4` just to realize the host is iOS. Surely this information can be gleaned from existing variables like `$target_os` or `$host`. Please use `AC_CASE` instead of plain shell `if` to implement this to match the rest of the `configure.ac`.